subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/server/http/HttpServer.groovy

```diff
@@ -104,6 +105,11 @@ class HttpServer extends ServerWithExpectations implements HttpServerFixture {
         }
     }
 
+    @Override
+    String toString() {
+        return "HttpServer " + String.valueOf(getUri())
```

Comment:
With this change `toString` can throw an assertion error if `server.started == false`. See the logic in `getUri`.

I agree a `toString` would be useful, but the `toString` shouldn't fail due to internal server state.
```suggestion
        if (server.started) {
            return "HttpServer " + String.valueOf(getUri())
        } else {
            return "HttpServer (unstarted)"
        }
```

Reply:
Thanks, good catch